src/lib/message-router.js

```diff
@@ -132,8 +132,10 @@ MessageRouter.prototype.receiveRoutes = function * (payload, sender) {
   let gotNewRoute = false
   for (const route of routes) {
     // We received a route from another connector, but that connector
-    // doesn't actually belong to the route, so ignore it.
+    // doesn't actually appear as the first hop of the route, so ignore it.
```

Comment:
Why did the comment change? It is still checking whether or not the route belongs to the connector that sent it.